packages/connectivity/src/scp-cf/destination/destination-accessor-provider-subscriber-lookup.spec.ts

```diff
@@ -296,3 +331,63 @@ describe('jwtType x selection strategy combinations. Possible values are {subscr
     });
   });
 });
+
+describe('call getAllDestinations with and without subscriber token', () => {
+  beforeEach(() => {
+    mockThingsForCombinations();
+  });
+
+  afterEach(() => {
+    nock.cleanAll();
+    jest.clearAllMocks();
+  });
+
+  const options: AllDestinationOptions = {
+    iasToXsuaaTokenExchange: false
+  };
+
+  it('should fetch all subscriber destinations', async () => {
+    const logger = createLogger({
+      package: 'connectivity',
+      messageContext: 'destination-accessor'
+    });
+    const debugSpy = jest.spyOn(logger, 'debug');
+
+    const allDestinations = await getAllDestinationsFromDestinationService({
+      ...options,
+      jwt: subscriberUserJwt
+    });
+
+    expect(allDestinations).toEqual([parsedSubscriberDestination]);
+
+    expect(debugSpy).toHaveBeenCalledWith(
+      'Retrieving all destinations for account: "subscriber" from destination service.'
+    );
+    expect(debugSpy).toHaveBeenCalledWith(
+      'Retrieving subaccount destination: DESTINATION.\n'
+    );
+  });
+
+  it('should fetch all provider destinations when called without anything', async () => {
+    const allDestinations = await getAllDestinationsFromDestinationService();
+    expect(allDestinations).toEqual([parsedProviderDestination]);
+  });
+
+  it('should not fetch anything', async () => {
+    nock.cleanAll();
+    jest.clearAllMocks();
+    mockThingsForCombinations(true, true);
+
+    const logger = createLogger({
+      package: 'connectivity',
+      messageContext: 'destination-accessor'
+    });
+    const warnSpy = jest.spyOn(logger, 'warn');
+
+    const allDestinations = await getAllDestinationsFromDestinationService();
+    expect(allDestinations).toEqual(null);
+    expect(warnSpy).toHaveBeenCalledWith(
+      'Could not retrieve destinations from destination service.'
+    );
+  });
```

Comment:
[req] Please add a test where the response of the destination service is service empty. See the specs (empty array) 
https://api.sap.com/api/SAP_CP_CF_Connectivity_Destination/resource. 

Also an error case like 404 or 500 would be nice to mock to ensure we give a resonable exception.

Reply:
I think with the `mockThingsForCombinations(true, true);` I already test the empty case, however, I guess I can add some 404/500 case.